tools/clang/unittests/HLSL/ExtensionTest.cpp

```diff
@@ -1281,6 +1289,31 @@ TEST_F(ExtensionTest, ResourceExtensionIntrinsic) {
   VERIFY_IS_TRUE(regex.match(disassembly));
 }
 
+ TEST_F(ExtensionTest, CustomLoadOp) {
+  Compiler c(m_dllSupport);
+  c.RegisterIntrinsicTable(new TestIntrinsicTable());
+  c.Compile(
+    "float2 main(uint2 v1 : V1) : SV_Target {\n"
+    "  return CustomLoadOp(uint2(1, 2));\n"
```

Comment:
I'm having trouble seeing how this code is exercising the new feature we are adding here, which is that we can now have functions that can have default values for parameters that are not given. So I would expect a test with two calls like

```
CustomLoadOp(uint2(1,2))
CustomLoadOp()
```

Where the first one gets the argument we pass and the second on gets the default (undef) value.

Reply:
Added optional argument test